src/ckeditor.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import getEditorNamespace from './utils/get-editor-namespace.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => type in { classic: 1, inline: 1 }
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.11.4/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	mounted() {
+		getEditorNamespace( this.editorUrl ).then( () => {
+			const config = { ...this.config };
+
+			// Overwrite config only if `component.readOnly` is set.
+			if ( this.readOnly !== null ) {
+				config.readOnly = this.readOnly;
+			}
+
+			const method = this.type === 'inline' ? 'inline' : 'replace';
+			const element = this.$el.firstElementChild;
+			const editor = this.instance = CKEDITOR[ method ]( element, config );
+
+			editor.on( 'instanceReady', () => {
+				const undo = editor.undoManager;
+				const data = this.value;
+
+				undo && undo.lock();
+
+				editor.setData( data, { callback: () => {
+					this.$_setUpEditorEvents();
+
+					const newData = editor.getData();
+
+					// Locking undoManager prevents 'change' event.
+					// Trigger it manually to update bound data.
+					if ( data !== newData ) {
+						this.$once( 'input', () => {
+							this.$emit( 'ready', editor );
+						} );
+
+						this.$emit( 'input', newData );
+					} else {
+						this.$emit( 'ready', editor );
+					}
+
+					undo && undo.unlock();
+				} } );
+			} );
+		} );
+	},
+
+	beforeDestroy() {
+		if ( this.instance ) {
+			this.instance.destroy();
+		}
+
+		// Note: By the time the editor is destroyed (editor.on( 'destroy' ) called)
+		// the Vue component will not be able to emit any longer. So emitting #destroy a bit earlier.
+		this.$emit( 'destroy', this.instance );
```

Comment:
I'm wondering if this event is correct :thinking: It emits `destroy` event in the context of editor instance, not the whole component. Although, I'm not sure if both should be distinguished for the end user and probably the event should be moved before destroying instance and renamed into `beforeDestroy`.

On the other hand, fixing iframe based editor should probably allow us to move this whole logic into the correct `destroy` component method. In that case, I would suggest refactoring this method to emit only `beforeDestroy` event so users can use it for some cleanup and emit correct `destroy` event when an editor is destroyed (from `component.destroy`). It also makes sense to check out if Vue is not calling similar events on its own (maybe by different API?). In that case, destroy events would be redundant.

Reply:
I'm more for removing this event at all. We can't emit real `destroy` event, replacing it with `beforeDestroy` feels a bit redundant, because whoever is using CKEditor can just use `beforeDestroy` hook of parent component. Also when needed `component.instance#destroy` event can be used.

Reply:
Mostly it opens more possibilities for cleanup for users. However, as you pointed out there is some workaround for it, so let's wait if the feature is implicit requested by the community.